lib/fastlane/plugin/revenuecat_internal/actions/get_latest_github_release_within_same_major_action.rb

```diff
@@ -0,0 +1,80 @@
+require 'fastlane/action'
+require 'fastlane_core/configuration/config_item'
+require 'fastlane_core/ui/ui'
+require_relative '../helper/revenuecat_internal_helper'
+
+module Fastlane
+  module Actions
+    class GetLatestGithubReleaseWithinSameMajorAction < Action
+      def self.run(params)
+        repo_name = params[:repo_name]
+        current_version = Gem::Version.new(params[:current_version])
+        UI.message("Getting latest release for #{repo_name} on GitHub within same major as #{current_version}")
+        tag_names = Helper::RevenuecatInternalHelper.get_github_release_tag_names(repo_name)
```

Comment:
@vegaro I modified this a bit in a few places. Some of the changes you might notice:
- Now we return the list of non-prerelease tag names from the helper and we do the selection here. 
- I also added an extra check to make sure there exists at least a version within the same major.
- I modified the environment variables used to pass parameters to the action to match the namespaces we use in the plugin.
- I added `type` and `optional` parameters to the action parameters, so they are consistent with the other actions in the plugin.
Do lmk if you have any questions or comments!

Reply:
Makes sense!